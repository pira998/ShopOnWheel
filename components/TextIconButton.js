import React from "react";

import {Animated,ScrollView,StyleSheet,TouchableWithoutFeedback,Modal ,View,TouchableOpacity,Image,Text } from "react-native";

import { COLORS,FONTS,SIZES,icons,constants } from "../constants";

const TextIconButton=({containerStyle,label,labelStyle,icon,iconStyle,onPress})=>{
    return(
        <TouchableOpacity
        style={{
            flexDirection:'row',
            alignItems:'center',
            justifyContent:'center',
            ...containerStyle
        }}
        onPress={onPress}>

            <Text style={{
                ...FONTS.body4,
                ...labelStyle
            }}>
                {label}
            </Text>

            <Image
            source={icon}
            style={{marginLeft:5,width:20,height:20,tintColor:COLORS.black,...iconStyle}}></Image>

        </TouchableOpacity>
    )
}

export default TextIconButton