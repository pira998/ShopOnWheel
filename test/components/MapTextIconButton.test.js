import React from 'react';
import renderer from 'react-test-renderer';
import TextIconButton from '../../components/MapTextIconButton.js'


test('renders correctly', () => {
  const tree = renderer.create(<TextIconButton />).toJSON();
  expect(tree).toMatchSnapshot();
});