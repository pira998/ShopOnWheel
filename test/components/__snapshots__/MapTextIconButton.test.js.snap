// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 22,
        "lineHeight": 30,
        "marginRight": 8,
      }
    }
  />
  <Image
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
</View>
`;
